python/TestHarness/RunParallel.py

```diff
@@ -156,7 +167,20 @@ def returnToTestHarness(self, job_index):
     did_pass = True
 
     output = 'Working Directory: ' + tester.specs['test_dir'] + '\nRunning command: ' + command + '\n'
-    output += self.readOutput(f)
+
+    if tester.specs.isValid('redirect_stdout'):
+      # If the tester enabled redirect_stdout and is using more than one processor
+      if tester.specs['redirect_stdout'] != False \
+         and (self.options.parallel != None and int(self.options.parallel) > 1) \
+         and int(tester.specs['max_parallel']) > 1:
```

Comment:
This logic is too complicated: You have a nested `if` statement, but both `else`'s have the same logic. Also you don't need to test against `max_parallel` explicitly. The `RunApp` Tester actually records the final parallel number to use after processing all restrictions so you can just look at that number and make a decision. So it boils down to a very simple `if` statement.